src/app/core/guards/page-not-found.guard.ts

```diff
@@ -19,30 +19,49 @@
 
 import {Injectable} from '@angular/core';
 import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot} from '@angular/router';
+
 import {I18n} from '@ngx-translate/i18n-polyfill';
 import {NotificationService} from '../notifications/notification.service';
+import {Observable} from 'rxjs/Observable';
+import {map} from 'rxjs/operators';
+import {DefaultWorkspaceModel} from '../store/users/user.model';
+import {selectCurrentUser} from '../store/users/users.state';
+import {AppState} from '../store/app.state';
+import {Store} from '@ngrx/store';
 
 @Injectable()
 export class PageNotFoundGuard implements CanActivate {
 
-  constructor(private i18n: I18n,
+  constructor(private store: Store<AppState>,
+              private i18n: I18n,
               private notificationService: NotificationService,
               private router: Router) {
   }
 
   public canActivate(next: ActivatedRouteSnapshot,
-                     state: RouterStateSnapshot): boolean {
+                     state: RouterStateSnapshot): Observable<boolean> {
     const [, w, organizationCode, projectCode] = state.url.split('/');
 
-    if (w === 'w' && organizationCode && projectCode) {
-      this.router.navigate(['w', organizationCode, projectCode, 'view', 'search']);
-    } else {
-      this.router.navigate(['workspace']);
-    }
+    return this.getDefaultWorkspace().pipe(
+      map(workspace => {
+        const hasWorkspace = workspace && workspace.organizationCode && workspace.projectCode;
+        if (w === 'w' && organizationCode && projectCode && hasWorkspace) {
+          this.router.navigate(['w', workspace.organizationCode, workspace.projectCode, 'view', 'search']);
```

Comment:
This behavior does not make sense to me. You check if both the default workspace and workspace from the URL is set and then redirect it to the default one.

I think it should make redirects in this order:

1. If workspace is defined in the URL, navigate to it.
2. If not and default workspace is defined, navigate to it.
3. Otherwise, navigate to workspace selection page.

What do you think?

Reply:
Ok, that make sense